pkg/mount/block/blockdev.go

```diff
@@ -387,32 +386,42 @@ func (b BlockDevices) FilterZeroSize() BlockDevices {
 	return nb
 }
 
-// FilterESP returns a list of BlockDev objects whose underlying block device
-// is a valid EFI system partition.
-func (b BlockDevices) FilterESP() BlockDevices {
-	return b.FilterGUID(SystemPartitionGUID.String())
-}
-
-// FilterGUID returns a list of BlockDev objects whose underlying
-// block device has the given GPT partition GUID.
+// FilterGUID returns a list of partition BlockDev objects whose partition GUID
```

Comment:
I'd reword a bit:
```diff
-// FilterGUID returns a list of partition BlockDev objects whose partition GUID
+// FilterGUID returns a list of partition BlockDev objects whose partition type GUID
```

`part.Type` on line 402 is the partition's type guid, not its unique partition guid.

Reply:
I propose renaming this function, perhaps to `FilterPartType()` and creating a new one `FilterPartID()` which compares `part.Id` rather than `part.Type`.

Reply:
good point. changing it.

Reply:
done!